pkg/monitor/monitor/util.go

```diff
@@ -448,32 +433,14 @@ func getMonitorGrafanaContainer(secret *core.Secret, monitor *v1alpha1.TidbMonit
 				Name:  "GF_PATHS_DATA",
 				Value: "/data/grafana",
 			},
-			{
-				Name: "GF_SECURITY_ADMIN_USER",
-				ValueFrom: &core.EnvVarSource{
-					SecretKeyRef: &core.SecretKeySelector{
-						LocalObjectReference: core.LocalObjectReference{
-							Name: secret.Name,
-						},
-						Key: "username",
-					},
-				},
-			},
-			{
-				Name: "GF_SECURITY_ADMIN_PASSWORD",
-				ValueFrom: &core.EnvVarSource{
-					SecretKeyRef: &core.SecretKeySelector{
-						LocalObjectReference: core.LocalObjectReference{
-							Name: secret.Name,
-						},
-						Key: "password",
-					},
-				},
-			},
```

Comment:
I suggest remain these logic for the backward compatibility for 1.1.x user. And add the comment that. We will finally drop these logic in 1.2.0

Reply:
I am not very sure. This is a serious security problem, do we really need to prioritize backward compatibility over security in this case?

Reply:
This is a good concern. I think we can break the backward compatibility here. Just mention this the in the request's release note.

Reply:
Especially in patch versions, we should not introduce break changes. The security problem is not unavoidable, we can encourage people to use the secret to store password after this.

We can remove `password` field in next minor version v1.12.

Reply:
Agree with @cofyc , we should maintain the backward compatibility here, and we can make it clear in the doc & comments that the secret should be used.